File: applications/plugins/org.csstudio.opibuilder.examples/examples/BOY Examples/scripts/SetIntensityGraphData.js
importPackage(Packages.java.lang);
importPackage(Packages.java.lang.reflect);
importPackage(Packages.org.csstudio.platform.data);
importPackage(Packages.org.eclipse.swt.graphics);

var simuData = java.lang.reflect.Array.newInstance(java.lang.Double.TYPE, 65536);
var value = ValueUtil.getDouble(pvArray[0].getValue());
var dataSrc = ValueUtil.getString(pvArray[1].getValue());

if(dataSrc == "Linear Sine Wave"){
	for(var i=0; i<256; i++){
		for(var j=0; j<256; j++){
			simuData[i*256 + j] = Math.sin(j*6*Math.PI/256 + i*6*Math.PI/256 + value);
		}
	}
}else{
	for(var i=0; i<256; i++){
		for(var j=0; j<256; j++){
			var x = j-128;
			var y = i-128;		
			var p = Math.sqrt(x*x + y*y);
			simuData[i*256 + j] = Math.sin(p*2*Math.PI/256 + value);		
		}
	}
}
widgetController.setValue(simuData);
	

File: applications/plugins/org.csstudio.opibuilder.examples/examples/BOY Examples/scripts/SetIntensityGraphData.py
from org.csstudio.opibuilder.scriptUtil import PVUtil

import array
import math

simuData = array.array('d', range(65536))
value = PVUtil.getDouble(pvs[0])
dataSrc = PVUtil.getString(pvs[1])

if dataSrc == "Linear Sine Wave":
	for i in range(256):
		for j in range(256):
			simuData[i*256 + j] = math.sin(j*6*math.pi/256 + i*6*math.pi/256 + value)
		
else:
	for i in range(256):
		for j in range(256):
			x = j-128
			y = i-128
			p = math.sqrt(x*x + y*y)
			simuData[i*256 + j] = math.sin(p*2*math.pi/256 + value)		
		
widget.setValue(simuData);	
